allennlp/nn/util.py

```diff
@@ -427,11 +430,21 @@ def sequence_cross_entropy_with_logits(logits: torch.FloatTensor,
     # shape : (batch * max_len, 1)
     targets_flat = targets.view(-1, 1).long()
 
-    # Contribution to the negative log likelihood only comes from the exact indices
-    # of the targets, as the target distributions are one-hot. Here we use torch.gather
-    # to extract the indices of the num_classes dimension which contribute to the loss.
-    # shape : (batch * sequence_length, 1)
-    negative_log_likelihood_flat = - torch.gather(log_probs_flat, dim=1, index=targets_flat)
+    if label_smoothing is not None and label_smoothing > 0.0:
+        num_classes = logits.size(-1)
+        smoothing_value = label_smoothing / num_classes
+        smoothed_targets = ones_like(log_probs_flat) * smoothing_value
+        # Fill all the correct indices with 1 - smoothing value.
```

Comment:
Update the logic to match the tensorflow code, like we discussed.  Also, it'd be nice to give a simple example in a comment (or the docstring above) of what this will do to the targets.